src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -121,6 +121,20 @@
       :active false)))
 
 
+(s/defn ^:private update-table-description!
+  "Update description for any CHANGED-TABLES belonging to DATABASE."
+  [database :- i/DatabaseInstance, changed-tables :- #{i/DatabaseMetadataTable}]
+  (log/info "Updating description for tables:"
+            (for [table changed-tables]
+              (sync-util/name-for-logging (table/map->TableInstance table))))
+  (doseq [{schema :schema, table-name :name, description :description} changed-tables]
+    (db/update-where! Table {:db_id       (u/get-id database)
+                             :schema      schema
+                             :name        table-name
+                             :description nil}
+                      :description description)))
```

Comment:
This looks like it will do an `UPDATE` for every single "changed" table regardless of whether it has a description or not.

On a first sync of a DB with 100,000 tables and 0 table comments that would result in an extra 100,000 DB calls like `UPDATE table SET description = NULL WHERE ...`. We *definitely* need to make sure we're not doing DB calls here to set a column that's already `NULL` to `NULL`.